tests/components/sensoterra/test_config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Test the Sensoterra config flow."""
+
+from enum import Enum
+from unittest.mock import AsyncMock, patch
+
+from jwt import DecodeError
+import pytest
+from sensoterra.customerapi import InvalidAuth as StInvalidAuth, Timeout as StTimeout
+
+from homeassistant.components.sensoterra.const import DOMAIN
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .const import API_EMAIL, API_PASSWORD, API_TOKEN, HASS_UUID, SOURCE_USER
+
+Test = Enum("Test", ["INVALID_AUTH", "TIMEOUT", "DECODE_ERROR"])
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    hass.data["core.uuid"] = HASS_UUID
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_EMAIL: API_EMAIL,
+            CONF_PASSWORD: API_PASSWORD,
+        },
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == API_EMAIL
+    assert result["data"] == {
+        CONF_TOKEN: API_TOKEN,
+        CONF_EMAIL: API_EMAIL,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.parametrize("test", [Test.INVALID_AUTH, Test.TIMEOUT, Test.DECODE_ERROR])
```

Comment:
Rather just add the exceptions you want to put in `side_effect` as parameter. We don't want if statements in the tests

Reply:
Yes, indeed, that was poorly written. Fixed.